doc/manual/strings.rst

```diff
@@ -230,12 +230,12 @@ a normal value:
 Using an index less than 1 or greater than ``end`` raises an error::
 
     julia> str[0]
-    ERROR: BoundsError()
-     in getindex at /Users/sabae/src/julia/usr/lib/julia/sys.dylib (repeats 2 times)
+    ERROR: BoundsError: attempt to access 14-element Array{UInt8,1} at index [0]
```

Comment:
I guess we should check the bounds from the string indexing code instead of passing them blindly to the underlying array. `@inbounds` would avoid doing the checks twice.